src/dialog/llm/__init__.py

```diff
@@ -20,5 +26,40 @@ def get_llm_class() -> Type[AbstractLLM]:
         llm_class = getattr(llm_module, class_name)
     except Exception as e:
         logging.info(f"Failed to load LLM class {Settings().LLM_CLASS}. Using default LLM. Exception: {e}")
+        raise
+
+    if isinstance(llm_class, AbstractLLM):
+        return llm_class, "AbstractLLM"
+
+    elif 'langchain_core.runnables' in str(type(llm_class)):
+        return llm_class, "LCELRunnable"
+
+    logging.info(f"Type for LLM class is: {type(llm_class)}")
+
+    return DialogLLM, "AbstractLLM"
+
+
+def process_user_message(message, chat_id=None):
+    LLM, llm_type = get_llm_class()
+    if llm_type == "AbstractLLM":
+        llm_instance = LLM(config=Settings().PROJECT_CONFIG, session_id=chat_id)
+        ai_message = llm_instance.process(message.message)
+
+    elif llm_type == "LCELRunnable":
+        ai_message = LLM.invoke(
+            {"input": message.message},
+            {"configurable": {
+                "session_id": chat_id,
+                "model": Settings().PROJECT_CONFIG.get("model_name", "gpt-3.5-turbo"),
+                **Settings().PROJECT_CONFIG,
+            }}
+        )
+        ai_message = {"text": ai_message.content}
+
+    return ai_message
```

Comment:
why does this function lives here in the `__init__.py`? If each class implementation have its own methods for processing, why not leave the chain with LCEL inside the `DialogLcelLLM` and make its parts more LCEL-like (prompt, retriever, memory etc)?

Reply:
This code was made in the __init__ so it's accessible through the module and on the API across different views. The theory is that we need to make thin views and fat models, but since we don't have too many models or business logic that runs on models, we should use this function as a controller, which process something and shrink the size of the view.